### PR TITLE
nox-review: Allow URLs to pull requests as well in 'nox-review pr'

### PR DESCRIPTION
**Pull request by dezgeg**
This allows users to be lazier and just copy the pull request URL
from the browser's address bar. That is, the following commands
do the same thing:

````
$ nox-review pr https://github.com/NixOS/nixpkgs/pull/12345
$ nox-review pr 12345
````

This also does the right thing on other repositories, avoiding the
need for the --slug option when an URL is given. E.g. these are again
equivalent:

````
$ nox-review pr https://github.com/mozilla/nixpkgs-mozilla/pull/58
$ nox-review pr --slug mozilla/nixpkgs-mozilla 58
````

Inspired by the 'hub' tool which allows similar syntax
('hub checkout https://github.com/NixOS/nixpkgs/pull/12345') to
easily checkout pull requests.

### CODE REVIEW

File: nox/review.py [hunk 1]
```diff
@@ -99,11 +100,17 @@ def wip(ctx, against):
 @click.option('--slug', default='NixOS/nixpkgs', help='The GitHub "slug" of the repository in the from of owner_name/repo_name.')
 @click.option('--token', help='The GitHub API token to use.')
 @click.option('--merge/--no-merge', default=True, help='Merge the PR against its base.')
-@click.argument('pr', type=click.INT)
+@click.argument('pr', type=click.STRING)
 @click.pass_context
 @setup_nixpkgs_config
 def review_pr(ctx, slug, token, merge, pr):
     """Build the changes induced by the given pull request"""
+    m = re.match('^(?:https?://(?:www\.)?github\.com/NixOS/nixpkgs/pull/)?([0-9]+)$', pr, re.IGNORECASE)
```
Comment (matthewbauer) [inline nox/review.py]: Could you modify this to set the slug option as well? So we can use something like this:

https://github.com/mozilla/nixpkgs-mozilla/pull/58

Comment (dezgeg) [inline nox/review.py]: I hadn't noticed that functionality. Thanks for noticing, now amended.

File: nox/review.py [hunk 2]
```diff
@@ -96,14 +97,30 @@ def wip(ctx, against):
...
 def review_pr(ctx, slug, token, merge, pr):
     """Build the changes induced by the given pull request"""
+
+    # Allow the 'pr' parameter to be either the numerical ID or an URL to the PR on GitHub.
+    # Also if it's an URL, parse the proper --slug argument from that.
+    m = re.match('^(?:https?://(?:www\.)?github\.com/([^/]+/[^/]+)/pull/)?([0-9]+)$', pr, re.IGNORECASE)
+    if not m:
+        click.echo("Error: parameter to 'nox-review pr' must be a valid pull request number or URL.")
+        sys.exit(1)
+    pr = m[2]
```
Comment (madjar) [inline nox/review.py]: Oh, I didn't know that match objects were indexable. That is neat!

**Comment by madjar** [comment]
That is awesome, thanks!

**Pull request merged**